tests/nested/core20/basic/task.yaml

```diff
@@ -8,26 +8,26 @@ execute: |
     . "$TESTSLIB/nested.sh"
 
     # wait for the system to be seeded first
-    execute_remote "sudo snap wait system seed.loaded"
+    nested_execute "sudo snap wait system seed.loaded"
 
     echo "Ensure 'snap install' works"
     # The install command could cause a ssh break, so || true is used
     # and then we check the installation was completed successfully
-    execute_remote "sudo snap install test-snapd-sh" || true
+    nested_execute "sudo snap install test-snapd-sh" || true
 
     echo "Ensure 'snap list' works and test-snapd-sh snap is installed"
-    execute_remote "snap list" | MATCH test-snapd-sh
+    nested_execute "snap list" | MATCH test-snapd-sh
 
     echo "Ensure 'snap find' works"
-    execute_remote "snap find test-snapd-sh" | MATCH ^test-snapd-sh
+    nested_execute "snap find test-snapd-sh" | MATCH ^test-snapd-sh
 
     echo "Ensure 'snap info' works"
-    execute_remote "snap info test-snapd-sh" | MATCH '^name:\ +test-snapd-sh'
+    nested_execute "snap info test-snapd-sh" | MATCH '^name:\ +test-snapd-sh'
 
     echo "Ensure 'snap remove' works"
     # The install command could cause a ssh break, so || true is used
     # and then we check the removal was completed successfully
-    execute_remote "sudo snap remove test-snapd-sh" || true
+    nested_execute "sudo snap remove test-snapd-sh" || true
 
     echo "Ensure 'snap list' works and test-snapd-sh snap is removed"
-    not execute_remote "snap list test-snapd-sh"
+    not nested_execute "snap list test-snapd-sh"
```

Comment:
Wait, how does this work? `nested_execute` is a function and `not` does not work with functions.

This is most certainly buggy and would probably work for:
```
not execute_remote true
```

Reply:
Please add a FIXME here, this is invalid use of `not`

Reply:
note added